test/functional/connection/connection.ts

```diff
@@ -359,5 +359,33 @@ describe("Connection", () => {
         })));
 
     });
+
+    describe("Can change postgres default schema name", () => {
+        let connections: Connection[];
+        beforeEach(async () => {
+            connections = await createTestingConnections({ 
+                enabledDrivers: ["postgres"],
+                entities: [Post]
+            });
+        });
+        afterEach(() => closeTestingConnections(connections));        
+        it("schema name can be set", () => {
+            return Promise.all(connections.map(async connection => {
+                await connection.syncSchema(true);
+
+                const post = new Post();
+                post.title = "ChangeSchemaName";
+
+                const PostRepo = connection.getRepository(Post);
+                await PostRepo.persist(post);
+
+                const query = await connection.driver.createQueryRunner();
+                const schemaName = connection.driver.options.schemaName || "public";
```

Comment:
looks like your test is completely wrong, you enable "postgres" instead of "postgres-schema" and here you use `|| "public"` which fixes this mistake, thats why your test is passing. it actually checks if query is executed in public schema. Please remove `schemaName` variable and use `"test-schema"."post"` right in the query.